trainer/src/trainer/upload_to_fireworks.py

```diff
@@ -0,0 +1,143 @@
+import json
+import os
+import logging
+import subprocess
+import time  # Added as per instructions
+
+
+def upload_to_fireworks(
+    fireworks_base_model: str,
+    lora_model_path: str,
+    fine_tune_id: str,
+):
+    """
+    Attempts to upload to fireworks with a specified timeout and number of retries.
+    :param fireworks_base_model: The base model for fireworks.
+    :param lora_model_path: The path to the Lora model.
+    :param fine_tune_id: The fine tune ID.
+    """
+
+    fireworks_json = {"base_model": fireworks_base_model}
+
+    with open(os.path.join(lora_model_path, "fireworks.json"), "w") as f:
+        json.dump(fireworks_json, f)
+
+    adapter_config_path = os.path.join(lora_model_path, "adapter_config.json")
+    with open(adapter_config_path, "r") as f:
+        adapter_config = json.load(f)
+        for key in ["use_rslora", "use_dora"]:  # Check for both keys
+            if key in adapter_config:
+                if adapter_config[key]:
+                    raise Exception(
+                        f"Fireworks does not support the {key} adapter config key"
+                    )
+                else:
+                    del adapter_config[key]
```

Comment:
Isn't this a no-op? If I read the code correctly, we throw an exception if this key is present, so deleting it shouldn't make any difference.

Reply:
Actually, we throw an error if the key is set to True. If it's false, we just delete it from the config (since Fireworks doesn't know what to do with it and throws an error if it's present).